config-files/init.go

```diff
@@ -0,0 +1,358 @@
+	}
+
+	if tableCount > 0 {
+		fmt.Println("Table `dynamodb_adapter_table_ddl` already exists. Skipping creation.")
+		return nil
+	}
+
+	op, err := adminClient.UpdateDatabaseDdl(ctx, &database.UpdateDatabaseDdlRequest{
+		Database:   db,
+		Statements: []string{ddl},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create table: %w", err)
+	}
+
+	return op.Wait(ctx)
+}
+
+func listDynamoTables(client *dynamodb.Client) ([]string, error) {
+	output, err := client.ListTables(context.TODO(), &dynamodb.ListTablesInput{})
+	if err != nil {
+		return nil, err
+	}
+	return output.TableNames, nil
+}
+
+func migrateDynamoTableToSpanner(ctx context.Context, db, tableName string, client *dynamodb.Client) error {
+	// Fetch attributes, partition key, and sort key from DynamoDB table
+	config, err := loadConfig("../config.yaml")
+	if err != nil {
+		log.Fatalf("Error loading configuration: %v", err)
+	}
+	models.SpannerTableMap[tableName] = config.Spanner.InstanceID
+	attributes, partitionKey, sortKey, err := fetchTableAttributes(client, tableName)
+	if err != nil {
+		return fmt.Errorf("failed to fetch attributes for table %s: %v", tableName, err)
+	}
+
+	// Generate Spanner index name and actual table name
+	//	spannerIndexName := fmt.Sprintf("index_%s", tableName)
+	actualTable := tableName
+
+	// Prepare mutations to insert data into the adapter table
+	var mutations []*spanner.Mutation
+	for column, dataType := range attributes {
+		mutations = append(mutations, spanner.InsertOrUpdate(
+			"dynamodb_adapter_table_ddl",
+			[]string{
+				"column", "tableName", "dataType", "originalColumn",
+				"partitionKey", "sortKey", "spannerIndexName", "actualTable",
+			},
+			[]interface{}{
+				column, tableName, dataType, column,
+				partitionKey, sortKey, column, actualTable,
+			},
+		))
+	}
+
+	// Perform batch insert into Spanner
+	if err := spannerBatchInsert(ctx, db, mutations); err != nil {
+		return fmt.Errorf("failed to insert metadata for table %s into Spanner: %v", tableName, err)
+	}
+
+	log.Printf("Successfully migrated metadata for DynamoDB table %s to Spanner.", tableName)
+	return nil
+}
+
+func fetchTableAttributes(client *dynamodb.Client, tableName string) (map[string]string, string, string, error) {
+	// Fetch table description
+	output, err := client.DescribeTable(context.TODO(), &dynamodb.DescribeTableInput{
+		TableName: aws.String(tableName),
+	})
+	if err != nil {
+		return nil, "", "", fmt.Errorf("failed to describe table %s: %w", tableName, err)
+	}
+
+	// Extract partition key and sort key
+	var partitionKey, sortKey string
+	for _, keyElement := range output.Table.KeySchema {
+		switch keyElement.KeyType {
+		case dynamodbtypes.KeyTypeHash:
+			partitionKey = aws.ToString(keyElement.AttributeName)
+		case dynamodbtypes.KeyTypeRange:
+			sortKey = aws.ToString(keyElement.AttributeName)
+		}
+	}
+
+	// Extract attributes from the table
+	attributes := make(map[string]string)
```

Comment:
Do we need to scan entire table? Should this be simply output.Table.AttributeDefinitions 

https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_TableDescription.html#DDB-Type-TableDescription-AttributeDefinitions